lib/mix/lib/mix/tasks/deps.compile.ex

```diff
@@ -261,9 +261,15 @@ defmodule Mix.Tasks.Deps.Compile do
     end
   end
 
-  defp do_command(%Mix.Dep{app: app, opts: opts}, config, command, print_app?, env \\ []) do
+  defp do_command(
+         %Mix.Dep{app: app, env_vars: env_vars, opts: opts},
```

Comment:
Please move `%Mix.Dep{app: app, opts: opts}` out of the function head and into the function body.